src/lib/portals/utils.ts

```diff
@@ -1,17 +1,266 @@
 import type { AssetId, ChainId } from '@shapeshiftoss/caip'
 import { ASSET_NAMESPACE, bscChainId, toAssetId } from '@shapeshiftoss/caip'
+import type { Asset, AssetsByIdPartial } from '@shapeshiftoss/types'
+import { createThrottle, isSome } from '@shapeshiftoss/utils'
 import axios from 'axios'
 import { getConfig } from 'config'
-import { CHAIN_ID_TO_PORTALS_NETWORK } from 'lib/market-service/portals/constants'
+import qs from 'qs'
+import { getAddress, isAddressEqual, zeroAddress } from 'viem'
+
+import { colorMap } from '../../lib/asset-service/service/colorMap'
+import generatedAssetData from '../../lib/asset-service/service/generatedAssetData.json'
+import { CHAIN_ID_TO_PORTALS_NETWORK } from './constants'
 import type {
   GetBalancesResponse,
   GetPlatformsResponse,
+  GetTokensResponse,
   PlatformsById,
   TokenInfo,
-} from 'lib/market-service/portals/types'
+} from './types'
+
+// Accommodate for script vs. web shenanigans
```

Comment:
i.e `getConfig()` *would* technically be available, but since web won't be running, won't work

Reply:
[question] Are these utils not intended for the client, wouldnt it be better to keep them separate to the client code? That way we arent adding code that is known not to work here

Reply:
Just to disambiguate things, this *is* client code atm, and moving it out of `src/lib` would still be client code (except if we move portals, and consequently asset gen to packages, which is absolute hell and we shouldn't do, spent a day trying to doso 💀). Client meaning within the web tsc project and not its separate package.

It is intended to (and is) consumed in both web running (as in, *running* web build) and in the context of scripts, (as in, web *project*, but web itself not running) meaning `getConfig()` isn't guaranteed to work. It should really, but...

Noticed two missing occurences of `getConfig()` however: 59d8d6586c

@woodenfurniture if this is confusing, happy to remove the paranoia short circuit, we really shouldn't need it!

Reply:
yes if we dont need it we should totally remove it